File: leetcode2791/xorTree.go
// func countPalindromePaths(parent []int, s string) int64 {

// 	n := len(parent)

// 	graph := make([][]int, n)

// 	for i := 1; i < n; i++ {

// 		graph[parent[i]] = append(graph[parent[i]], i)

// 	}

// 	ans := int64(0)

// 	cnt := map[int]int{0: 1}

// 	var dfs func(int, int)

// 	dfs = func(v int, xor_ int) {

// 		for _, w := range graph[v] {

// 			x := xor_ ^ (1 << (s[w] - 'a'))

// 			ans += int64(cnt[x])

// 			for i := 0; i < 26; i++ {

// 				ans += int64(cnt[x^(1<<i)])

// 			}

// 			cnt[x]++

// 			dfs(w, x)

// 		}

// 	}

// 	dfs(0, 0)

// 	return ans

// }

File: leetcode2791/xorTree.cpp
// class Solution 
// {
// public:
//     long long countPalindromePaths(vector<int>& parent, string s) 
//     {
//         const int n=parent.size();
//         int i;

//         vector<vector<int>>graph(n);
//         for(i=1;i<n;++i)graph[parent[i]].emplace_back(i);

//         long long ans=0;
//         unordered_map<int,int>cnt{{0,1}};

//         function<void(int,int)>dfs=[&](int v,int xor_)
//         {
//             for(const auto&w:graph[v])
//             {
//                 int x=xor_^(1<<(s[w]-'a'));
//                 ans+=cnt.contains(x)?cnt[x]:0;
//                 for(int i=0;i<26;++i)
//                     ans+=cnt.contains(x^(1<<i))?cnt[x^(1<<i)]:0;
            
//                 ++cnt[x];
//                 dfs(w,x);
//             }
//         };

//         dfs(0,0);


//         return ans;
//     }
// };